compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -356,7 +356,11 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
 
     val map = new TypeMap {
       def apply(tp: Type) = tp match {
-        case tref: TypeRef if isPatternTypeSymbol(tref.typeSymbol) => WildcardType(tref.underlying.bounds)
+        case tp @ AppliedType(tycon, args) if tycon.isRef(defn.ArrayClass) =>
+          // walkaround `Array[_] <:< Array[Array[_]]`, see tests/patmat/t2425.scala
```

Comment:
I don't think this comment is true, the following code doesn't compile:
```scala
def foo(x: Array[_]): Array[Array[_]] = x
```

Reply:
The underscore above expands to TypeBounds `Array[<: Any :> Nothing]` instead of Wildcards. 

Here is the subtyping log:

```
  ==> Array[(?)] <:< Array[Array[(?)]]
    ==> scala.type <:< scala.type
    <== scala.type <:< scala.type   = true
    ==> Array[(?)] <:< (?)
      ==> Any <:< Any LoApprox
      <== Any <:< Any LoApprox  = true
    <== Array[(?)] <:< (?)   = true
    ==> (?) <:< Array[(?)]
    <== (?) <:< Array[(?)]   = true
  <== Array[(?)] <:< Array[Array[(?)]]   = true
```

And the behavior can be confirmed from the code in TypeComparer.

But thanks for raising the question, as by using TypeBounds instead WildcardType, we avoid the problem.

Reply:
Yes, `WildcardType(lo, hi) <:< X` is equivalent to `lo <:< X`, see TypeComparer